Document file system object content addressing (#10479)
NixOS/nix: doc/manual/src/store/store-path.md

> store directory            digest                 name
> ```

Exactly how the digest is calculated depends on the type of store path.

> Store path digests are *supposed* to be opaque, and so for most operations, knowing how the digest is computed is not necessary.

Reviewer: ```suggestion
Store path digests are *supposed* to be opaque, and so for most operations, it is not necessary to know the details.
```

> That said, at the end of the manual is a full [specification](@docroot@/protocols/store-path.md) of store path digests.

Reviewer: ```suggestion
That said, the manual has a full [specification of store path digests](@docroot@/protocols/store-path.md).
```

We're in hyperspace.
